test/integration/test_options.py

```diff
@@ -127,3 +131,19 @@ def test_optimization_and_resilience_levels(self, service):
                     cls_default = cls(options=user_given_options)
                     self.assertTrue(cls_default.options.optimization_level == opt_level)
                     self.assertTrue(cls_default.options.resilience_level == opt_level)
+
+        for primitive in primitives:
+            primitive_simulator = primitive(session=backend)
+            self.assertEqual(primitive_simulator.options.resilience_level, 0)
+            self.assertEqual(primitive_simulator.options.optimization_level, 1)
+
+            if real_backend:
+                primitive_real_backend = primitive(session=real_backend)
+                self.assertEqual(
+                    primitive_real_backend.options.resilience_level,
+                    Options._DEFAULT_RESILIENCE_LEVEL,
+                )
```

Comment:
Since we have a [fake runtime client](https://github.com/Qiskit/qiskit-ibm-runtime/blob/main/test/unit/mock/fake_runtime_client.py), a better test would be to ensure the value sent to the "server" has the right level, and not just what's recorded locally. This would also avoid having to run on a real backend.

Reply:
You might want to have a look at the tests I wrote for this purpose - I split the test `test_optimization_and_resilience_levels` into two to cover more cases. See 
https://github.com/Qiskit/qiskit-ibm-runtime/pull/752/files#diff-9f1396626262723a7bef9a07fc75f30acccb1dd18cd8ea41498a7f050a265563